plugins/eni/commands/commands.go

```diff
@@ -152,6 +163,63 @@ func add(args *skel.CmdArgs, engine engine.Engine, dhclient engine.DHClient) err
 	return cnitypes.PrintResult(result, conf.CNIVersion)
 }
 
+// getMACAddressOfENIAndVerify acquires the mac address of eni and verify the ip address information
+func getMACAddressOfENIAndVerify(conf *types.NetConf, engine engine.Engine, timeout time.Duration) (string, error) {
+	retriever := &ec2MetadataRetriever{
+		timeout: timeout,
+		conf:    conf,
+		engine:  engine,
+	}
+
+	mac, err := retriever.retrieveMacAddress()
+	if err != nil {
+		return "", err
+	}
+
+	err = validateIPAddressOfENI(conf, mac, engine)
+	if err != nil {
+		return "", err
+	}
+
+	return mac, nil
+}
+
+// ec2MetadataRetriever is used to retrieve the information of the eni from ec2 instance metadata
+type ec2MetadataRetriever struct {
+	macAddress string
+	timeout    time.Duration
+	conf       *types.NetConf
+	engine     engine.Engine
+}
+
+func (retriever *ec2MetadataRetriever) retrieveMacAddress() (string, error) {
+	backoff := utils.NewSimpleBackoff(ec2InstanceMetadataBackoffMin, ec2InstanceMetadataBackoffMax, ec2InstanceMetadataBackoffJitter, ec2InstanceMetadataBackoffMultiple)
+	ctx, cancel := context.WithTimeout(context.TODO(), retriever.timeout)
+	defer cancel()
+
+	err := utils.RetryWithBackoffCtx(ctx, backoff, func() error {
+		mac, err := getMACAddressOfENI(retriever.conf, retriever.engine)
+		if err == nil {
+			retriever.macAddress = mac
+			return nil
+		}
+
+		_, ok := err.(*engine.UnmappedMACAddressError)
```

Comment:
I realize that we don't actually have the issue here, but this style of type assertion gives me flashbacks to https://github.com/aws/amazon-ecs-agent/pull/910. [Interface conversions](https://golang.org/doc/effective_go.html#interface_conversions) are a less error prone way to perform type assertions, IMO.

Reply:
Can you address @nmeyerhans 's comment here as well?